src/downloader/downloader.hpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _DOWNLOADER_HPP_
+#define _DOWNLOADER_HPP_
+
+#include <functional>
+#include <string>
+
+#include <Poco/TaskManager.h>
+#include <Poco/URI.h>
+
+#include <aos/common/tools/error.hpp>
+
+class Downloader {
+public:
+    using FinishedCallback = std::function<void(const std::string&, aos::Error)>;
+
+    Downloader(const std::string& downloadDir);
+    ~Downloader();
+
+    aos::RetWithError<std::string> DownloadSync(const std::string& url);
+    void                           DownloadAsync(const std::string& url, FinishedCallback callback);
+
+private:
+    constexpr static std::chrono::milliseconds kDelay {1000};
```

Comment:
```suggestion
    constexpr static std::chrono::milliseconds cDelay {1000};
```
Constant should start with 'c' prefix

Reply:
Done